components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -872,6 +872,12 @@ public String getOASDefinitionForPublisher(API api, String oasDefinition) throws
             for (Map.Entry<HttpMethod, Operation> entry : operationMap.entrySet()) {
                 Operation operation = entry.getValue();
                 operation.setVendorExtension(APIConstants.X_WSO2_APP_SECURITY, appSecurityExtension);
+                if (!operation.getVendorExtensions().containsKey(APIConstants.SWAGGER_X_THROTTLING_LIMIT)) {
+                    // set default throttling limit if not defined for operation in openapi definition. This is expected
+                    // from the choreo console.
+                    operation.setVendorExtension(APIConstants.SWAGGER_X_THROTTLING_LIMIT,
```

Comment:
Just to clarify:
If there's an existing operation with `50KPerMin` don't we need to convert that to the `x-throttling-limit` value (IMO we can use the existing method to get that value. I.E method with the switch case)?
+
Can we update the comment in line 866 ?

Reply:
good point, yeah we need to fix it in that way. If x-throttling-tier is available then we need to populate this limit accordingly.